### ISSUE TITLE
new behavior darp/subscribe is called every 15 seconds resulting in flood of 404

### ISSUE THREAD

**Issue by bgelens** [dapr/dapr#5972]
## In what area(s)?

 /area runtime

we upgraded to dapr 1.10.* and noticed a new behavior. It used to be that the endpoint `/darp/subscribe` would only be called during startup. Now it is being called repeatedly. The reason we notice is that for some of our services which do have a dapr sidecar but don't call the SDKs (in our case dotnet) `MapSubscribeHandler`, will generate a flood of 404 exceptions in our telemetry (application insights). This used to be only a single exception at startup.

The reason the endpoint is not created, is because the applications don't expose any endpoints to subscribe to (deliberate choice)

> /area operator

> /area placement

> /area docs

> /area test-and-release

## What version of Dapr?

1.10.x
> 1.1.x
> 1.0.x
> edge: output of `git describe --dirty`

## Expected Behavior

Old behavior, single call to this endpoint at startup and stop when 404 is returned

## Actual Behavior

Every 15 seconds the endpoint is called resulting in a 404 from our application

## Steps to Reproduce the Problem

Have a pubsub component. Do not map subscribe endpoint

## Release Note

RELEASE NOTE:

**Comment by ItalyPaleAle** [comment]
There's a built-in resiliency policy that makes this operation retry in case of failures, but it should only be retried 3 times (so for a total of 4 times). Does the retry happen indefinitely?

PS: What happens if you create a `/dapr/subscribe` endpoint which returns an empty list of subscriptions?

**Comment by olitomlinson** [comment]
> There's a built-in resiliency policy that makes this operation retry in case of failures, but it should only be retried 3 times (so for a total of 4 times). Does the retry happen indefinitely?
> 
> PS: What happens if you create a `/dapr/subscribe` endpoint which returns an empty list of subscriptions?

Hmmm… I’m using an out of the box install of 1.10 and with dapr run, I only ever get 1 call to dapr/subscribe, regardless of the status code response. I even raised it here yesterday on Discord https://discord.com/channels/778680217417809931/778684372475707416/1077644580608024687

**Comment by bgelens** [comment]
>There's a built-in resiliency policy that makes this operation retry in case of failures, but it should only be retried 3 times (so for a total of 4 times). Does the retry happen indefinitely?

@ItalyPaleAle Yes it happens continuously. I had a thought it might be caused by our api healthchecks that are used by start, ready, liveliness. We do have one that calls into dapr healthcheck. However, I don't see any failure reported on k8s end which I would expect when something comes from this route.

>PS: What happens if you create a /dapr/subscribe endpoint which returns an empty list of subscriptions?

All the other apps do have the endpoint and do not have this behavior.

I'll see if I can build a repo in isolation to allow for more investigation (I can't commit on a timeline for this)

@olitomlinson thanks for pointing me to the discord server! I didn't know about it yet so I just joined the party :)
From what I can tell the diff in our situration is 5xx vs 4xx

**Comment by bgelens** [comment]
@ItalyPaleAle @olitomlinson I tried to repro the issue today in isolation and I couldn't. The behavior was exactly the same as the "old" behavior that I described. A one time 404 at startup on the application and then it was over, just an empty subscriptions collection logged in the sidecar `app responded with subscriptions []`. I tried with 1.10.2, 1.10.1 and 1.10.0 just to be sure it was nothing in between fix releases. All worked fine.

So I started to think about what else could cause this and I only could think about 1 thing that is rolled out in our dev environment which is not on the rest of our clusters and that is Conductor. So I scaled down the agent and put a faulty node selecter in the daemonset and that was it. No more exceptions.

Tagging @msfussell as he probably wants to route this to the correct team (I'm not sure where to log this elsewhere)

Thanks all for the quick responses!
